.github/workflows/publish-docker-image.yml

```diff
@@ -2,57 +2,146 @@ name: Release multi-arch Docker Image
 
 on:
   release:
-    types: [published]
+    types: [ published ]
 
 jobs:
-  release:
+  release-linux:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
-
-    - uses: actions/setup-go@v5
-      with:
-        go-version: '1.21'
-    - run: go version
-
-    - name: Login to Docker Hub
-      uses: docker/login-action@v1
-      with:
-        username: ${{ secrets.DOCKERHUB_USERNAME }}
-        password: ${{ secrets.DOCKERHUB_TOKEN }}
-
-    - name: go-mod verify
-      run: go mod verify
-
-    - name: go-mod tidy
-      run: go mod tidy
-
-    - name: go-build
-      env:
-        GOOS: linux
-      run: go build
-
-    - name: go-test
-      run: go test -v ./...
-
-    - name: Set up QEMU
-      uses: docker/setup-qemu-action@v3
-
-    - name: Set up Docker Buildx
-      id: buildx
-      uses: docker/setup-buildx-action@v2
-
-    - name: Docker info
-      run: docker info
-    - name: Buildx inspect
-      run: docker buildx inspect
-
-    - name: Build and push image
-      uses: docker/build-push-action@v3
-      with:
-        context: .
-        file: ./Dockerfile
-        platforms: linux/amd64,linux/arm/v7,linux/arm64,linux/ppc64le,linux/s390x
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
```

Comment:
Same here: https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file